Inject preconstructed ApplicationContext in Lambda handler

ttzn opened the pull request:
This helps solving https://github.com/micronaut-projects/micronaut-test/issues/330. It basically allows initializing `MicronautRequestHandler`s with the app context from a `@MicronautTest` in this fashion:

```kotlin
@MicronautTest(contextBuilder = [LambdaApplicationContextBuilder::class])
class RequestHandlerTest {
    @Inject
    lateinit var context: ApplicationContext

    @Test
    fun testHandler() {
        val sampleRequestHandler = SampleRequestHandler(context)
        ...
```

This is probably still too much work on the user's part. Before going further I'd like this to be validated first.

graemerocher commented:
Looks like a good start. It would probably be work creating a stereotype annotations like:

```java
@MicronautTest(contextBuilder = [LambdaApplicationContextBuilder::class])
@interface MicronautLambdaTest {

}
```

It could also maybe provide injection of the handler so the user doesn't have to do that by adding additional extensions .

ttzn commented:
@graemerocher, a dedicated annotation is what I had in mind. However since it's not possible to "redirect" parameters from an annotation to a meta-annotation, I'll have to basically copy-paste the whole `@MicronautTest` body so that the user can still customize the application context as they please.

As for injecting handlers, the most user-friendly way would be to annotate them with `@Singleton`, but when I tried that I ran into https://github.com/micronaut-projects/micronaut-core/issues/4586 (I made an MR for that).

ttzn commented:
@graemerocher I added a `MicronautLambdaTest` annotation with a supporting JUnit 5 extension. Two points to consider:

* I added a new `micronaut-function-aws-test` subproject to isolate this feature;
* there is quite a bit of code that is replicated from `MicronautJunit5Extension`, a separate MR in `micronaut-test` would allow optimal code reuse if you agree with these changes.

Also maybe I should rename this branch as it has outgrown its initial purpose.

graemerocher commented:
Thanks, we probably need documentation for the new module and will need to bump the minor version of the project before merging this PR

ttzn commented:
I added documentation. It would be best if https://github.com/micronaut-projects/micronaut-test/pull/336 were merged first because it would allow adding the `contextBuilder` option and thus feature-parity with `@MicronautTest`.

graemerocher commented:
Thanks for the contribution
